interp/notation_ops.mli

```diff
@@ -85,22 +88,28 @@ exception No_match
 
 val print_parentheses : bool ref
 
+type 'a with_vars = Id.Set.t * (Name.t * Id.t) list * 'a
+
+(** Instances of notation variables, with their type as interpreted in
+    a term; also, when a variable is used both as term and binder, it
+    is represented as a a pattern *)
+type 'a glob_constr_notation_substitution =
+  ('a glob_constr_g with_vars, 'a cases_pattern_disjunction_g with_vars, 'a extended_glob_local_binder_g list with_vars) notation_arg_type
+    Id.Map.t
+
 val match_notation_constr : print_univ:bool -> 'a glob_constr_g -> vars:Id.Set.t -> interpretation ->
-      ((Id.Set.t * 'a glob_constr_g) * extended_subscopes) list *
-      ((Id.Set.t * 'a glob_constr_g list) * extended_subscopes) list *
-      ((Id.Set.t * 'a cases_pattern_disjunction_g) * extended_subscopes) list *
-      ((Id.Set.t * 'a extended_glob_local_binder_g list) * extended_subscopes) list
+  'a glob_constr_notation_substitution
+
+(** Instances of notation variables, with their type as interpreted in
+    a pattern *)
+type 'a glob_cases_pattern_notation_substitution =
+  ('a glob_constr_g, 'a cases_pattern_g, unit) notation_arg_type
+    Id.Map.t
 
 val match_notation_constr_cases_pattern :
   'a cases_pattern_g -> interpretation ->
-  (('a cases_pattern_g * extended_subscopes) list *
-   ('a cases_pattern_g list * extended_subscopes) list *
-   ('a cases_pattern_g * extended_subscopes) list) *
-    (bool * int * 'a cases_pattern_g list)
+  'a glob_cases_pattern_notation_substitution * (bool * int * 'a cases_pattern_g list)
```

Comment:
We could probably take the opportunity to document this (what are the boolean and the integer for isntance?)